lib/corebulk.go

```diff
@@ -288,7 +298,10 @@ func (b *BulkIndexer) startDocChannel() {
 			}
 			b.mu.Unlock()
 		}
+		close(doneChan)
 	}()
+
+	return doneSignal
```

Comment:
Could you just return doneChan here?  I think it would cast it as directional.

Reply:
Yeah, that would work. I wanted to use a named return value so the purpose was obvious, but it felt silly to have a named return value that I never assign a value to anywhere in the function and that I don't actually use when returning.

Reply:
Make sense.  FWIW, I leave them unassigned all the time.  ;)  I tend to feel the decision to name them should be independent of how the code is organized.  Maybe it's best done in combination with the `nakedret` linter, which ensures that we don't ever just call `return`.